exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/WorkspaceSchemaFactory.java

```diff
@@ -175,6 +193,21 @@ public WorkspaceSchema createSchema(List<String> parentSchemaPath, SchemaConfig
     return new WorkspaceSchema(parentSchemaPath, schemaName, schemaConfig);
   }
 
+  public WorkspaceSchema createSchema(List<String> parentSchemaPath, SchemaConfig schemaConfig, DrillFileSystem fs) throws IOException {
+    if (!accessible(fs)) {
```

Comment:
Is returning null sufficient to tell the user that they don't have permission to do this operation?

Reply:
returning null then user could not even list this workspace, so they don't know the existence of this workspace at all. I think that is a good access control practice. 

If users expect to see a workspace but could not see it, then they need to figure out why by themselves.